app/models/user.rb

```diff
@@ -98,6 +98,25 @@ def token
     JsonWebToken.encode(user_id: self.id, roles: [], email: self.email, verified: verified)
   end
 
+  def self.from_social(data)
+    Rails.logger.info "************ email is = #{data[:email]}"
+    user = User.where(email: data[:email]).first
+
+    path = '/profile'
+    unless user
+      Rails.logger.info "!!!!!! path is = #{path}"
+      user = User.new(first_name: data[:first_name],
+        last_name: data[:last_name],
+        email: data[:email],
+        zip: data[:zip],
+        password: data[:password]
+      )
+      path = '/signup-info'
+    end
+
+    arr = [user, path]
```

Comment:
You can remove this `arr` variable assignment.  Ruby will actually return the last line, for you.

So you can change it to just:

```ruby
[user, path]
```

Reply:
done.